config/jest.config.js

```diff
@@ -33,11 +34,12 @@ module.exports = {
     '/node_modules/(?!@edx)',
   ],
   transform: {
-    '^.+\\.[t|j]sx?$': [
```

Comment:
[curious] Is `babel-jest` supposed to transform `.js` files or only `.jsx` files?

Reply:
According to the prior config, only .jsx files, so I kept it that way.  That said, I don't see any .js/.ts unit tests, so it might be good to add some and make sure this configuration handles them.

Reply:
Yeah, I agree. Feel free to defer on this for another PR and/or file an issue since its unrelated to the core TypeScript changes here, but also cool to include it here while we're thinking about it :) Up to you haha.